frontend/src/pages/RunDetailsV2.test.tsx

```diff
@@ -395,6 +396,22 @@ describe('RunDetailsV2', () => {
 
       expect(screen.getAllByText('-').length).toEqual(2); // finish time and duration are empty.
     });
+
+    it('switches to Pipeline Spec tab', async () => {
+      render(
+        <CommonTestWrapper>
+          <RunDetailsV2
+            pipeline_job={v2YamlTemplateString}
+            runDetail={TEST_RUN}
+            {...generateProps()}
+          ></RunDetailsV2>
+        </CommonTestWrapper>,
+      );
+
+      userEvent.click(screen.getByText('Pipeline Spec'));
+      screen.findByTestId('spec-ir');
```

Comment:
Whenever possible, we should avoid testing visual element instead of test id. Can we detect the existence of `some-template-string` instead of `spec-ir`?